zallet/src/components/json_rpc/methods/get_raw_transaction.rs

```diff
@@ -0,0 +1,751 @@
+
+    let (vin, vout) = match tx.transparent_bundle() {
+        Some(bundle) => (
+            bundle
+                .vin
+                .iter()
+                .map(|tx_in| TransparentInput::encode(tx_in, bundle.is_coinbase()))
+                .collect(),
+            bundle
+                .vout
+                .iter()
+                .zip(0..)
+                .map(TransparentOutput::encode)
+                .collect(),
+        ),
+        _ => (vec![], vec![]),
+    };
+
+    let (vjoinsplit, join_split_pub_key, join_split_sig) = match tx.sprout_bundle() {
+        Some(bundle) if !bundle.joinsplits.is_empty() => (
+            bundle.joinsplits.iter().map(JoinSplit::encode).collect(),
+            Some(TxId::from_bytes(bundle.joinsplit_pubkey).to_string()),
+            Some(hex::encode(bundle.joinsplit_sig)),
+        ),
+        _ => (vec![], None, None),
+    };
+
+    let (value_balance, value_balance_zat, v_shielded_spend, v_shielded_output, binding_sig) =
+        if let Some(bundle) = tx.sapling_bundle() {
+            (
+                Some(value_from_zat_balance(*bundle.value_balance())),
+                Some(bundle.value_balance().into()),
+                Some(
+                    bundle
+                        .shielded_spends()
+                        .iter()
+                        .map(SaplingSpend::encode)
+                        .collect(),
+                ),
+                Some(
+                    bundle
+                        .shielded_outputs()
+                        .iter()
+                        .map(SaplingOutput::encode)
+                        .collect(),
+                ),
+                Some(hex::encode(<[u8; 64]>::from(
+                    bundle.authorization().binding_sig,
+                ))),
+            )
+        } else {
+            (None, None, None, None, None)
+        };
+
+    let orchard = tx
+        .version()
+        .has_orchard()
+        .then(|| Orchard::encode(tx.orchard_bundle()));
+
+    Ok(serde_json::value::to_raw_value(&Transaction {
+        in_active_chain: None,
+        hex: tx_hex,
+        txid: txid_str.to_ascii_lowercase(),
+        authdigest: TxId::from_bytes(tx.auth_commitment().as_bytes().try_into().unwrap())
+            .to_string(),
+        size,
+        overwintered,
+        version: tx.version().header() & 0x7FFFFFFF,
+        versiongroupid: overwintered.then(|| format!("{:08x}", tx.version().version_group_id())),
+        locktime: tx.lock_time(),
+        expiryheight: overwintered.then(|| tx.expiry_height().into()),
+        vin,
+        vout,
+        vjoinsplit,
+        value_balance,
+        value_balance_zat,
+        v_shielded_spend,
+        v_shielded_output,
+        binding_sig,
+        orchard,
+        join_split_pub_key,
+        join_split_sig,
+        blockhash,
+        height: None,
+        confirmations: None,
+        time: None,
+        blocktime: None,
```

Comment:
Currently does not return any mined-in-block information.

Reply:
I'm now copying this from the current underlying `getrawtransaction` RPC call, but should be able to fetch this correctly once we migrate to `ChainIndex`.